src/models/httpRequest.ts

```diff
@@ -1,14 +1,16 @@
 "use strict";
 
 import { Stream } from 'stream';
+import { RequestVariableCacheKey } from './requestVariableCacheKey';
 
 export class HttpRequest {
     public constructor(
         public method: string,
         public url: string,
         public headers: { [key: string]: string },
         public body: string | Stream,
-        public rawBody: string) {
+        public rawBody: string,
+        public requestVariableCacheKey?: RequestVariableCacheKey) {

```

Comment:
It seems we won't pass this in `constructor` call, how about move this to a class instance field, which should be supported by TypeScript:
```ts
export class HttpRequest {
    public requestVariableCacheKey?: RequestVariableCacheKey;
    public constructor(
        public method: string,
        public url: string,
        public headers: { [key: string]: string },
        public body: string | Stream,
        public rawBody: string) {
    }
```